javascripts/discourse/components/toc-contents.gjs

```diff
@@ -50,6 +51,14 @@ export default class TocContents extends Component {
     window.addEventListener("resize", this.calculateHeadingPositions);
   }
 
+  @action
+  listenForPostChange() {
```

Comment:
Any app event you can create has to be removed with `off` when the component is destroyed or when you are done with it.